Data Structures and Algorithms/Data Structures and Algorithms/Queue/QueueArray.swift

```diff
@@ -36,3 +36,15 @@ extension QueueArray: CustomStringConvertible {
         return String(describing: storage)
     }
 }
+
+// MARK: - Challenge 4 method
+extension QueueArray {
+    public mutating func reversed() -> QueueArray {
+        var reversedQueue = QueueArray()
+        
+        while !self.isEmpty {
+            reversedQueue.enqueue(storage.removeLast())
+        }
+        return reversedQueue // return a copy of the reversed queue
```

Comment:
are you sure this is a copy? the removeLast does remove the value from the storage so at the end your storage will be empty

Reply:
Solved, using a copy of the QueueArray, and removing the last from that copy.